gc.c

```diff
@@ -225,7 +225,46 @@ int sexp_valid_object_p (sexp ctx, sexp x) {
 #define sexp_gc_pass_ctx(x)
 #endif
 
-void sexp_mark_one (sexp_gc_pass_ctx(sexp ctx) sexp* types, sexp x) {
+struct sexp_mark_stack_ptr_t {
+  sexp *start, *end;
+  struct sexp_mark_stack_ptr_t *next; /* TODO: remove for allocations on stack */
+};
+struct sexp_mark_stack_ptr_t *sexp_mark_stack_ptr = NULL;
+
+/* TODO: make resizable */
+#define SEXP_MARK_STACK_COUNT 1024
+struct sexp_mark_stack_ptr_t sexp_mark_stack[SEXP_MARK_STACK_COUNT];
```

Comment:
These variables should be a context global, or part of the heap.  Chibi supports multiple independent VMs running in parallel.

Reply:
Ok, I will try to use context to store stack there